### PR TITLE
Improve Migration Guide for YoutubeAudioSourceManager

### PR DESCRIPTION
**Pull request by adrianojsoares1**
**Type of Change**
Documentation Update

**Description**
I wanted to introduce some more context into migration from the deprecated `YoutubeAudioSourceManager` in the Lavaplayer repository. It may not be clear to developers that this class is still being instantiated by default, and that it should be excluded.

This is related to changes made to Lavaplayer in release [2.1.2](https://github.com/lavalink-devs/lavaplayer/releases/tag/2.1.2)

The following image demonstrates what happens when the supported `YoutubeAudioSourceManager` is registered without excluding the deprecated client. There are now two instances of the client, which could lead to unexpected behavior.

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -177,7 +177,26 @@ Currently, the following clients are available for use:
...
+explicitly excluded.
+
+To avoid this unwanted instantiation, leverage the `excludeSources` parameter during remote source registration.
+```java
+AudioPlayerManager playerManager = new DefaultAudioPlayerManager();
+AudioSourceManagers.registerRemoteSources(playerManager,
+                                          com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.class);
+```
+
+Then, create and register an instance of the supported `YoutubeAudioSourceManager` from the `youtube-source` package.
```
Comment (devoxin) [inline README.md:193]: Move this section above the first codeblock. The new source manager should be registered before the others, as `registerRemoteSources` includes the `HttpAudioSourceManager` which catches all identifiers passed to it, meaning that the youtube source would never get queried when used like this.

File: README.md [hunk 2]
```diff
@@ -177,7 +177,26 @@ Currently, the following clients are available for use:
     client enabled, age-restricted tracks are **not** guaranteed to play.
 
 ## Migration from Lavaplayer's built-in YouTube source
-This client is intended to be a drop-in replacement, however there are a couple of things to note:
+
+This client is intended as a direct replacement for Lavaplayer's `YoutubeAudioSourceManager`, which the maintainers
+deprecated in release [2.1.2](https://github.com/lavalink-devs/lavaplayer/releases/tag/2.1.2).
```
Comment (devoxin) [inline README.md:182]: I wouldn't include a hard reference to this specific deprecation as other forks may not necessarily have deprecated the source, or they might've deprecated it separately as is the case with my fork.

**Review by devoxin — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -177,7 +177,26 @@ Currently, the following clients are available for use:
...
+
+This client is intended as a direct replacement for Lavaplayer's `YoutubeAudioSourceManager`,
+which may be deprecated in more recent releases of Lavaplayer.
+By default, Lavaplayer instantiates and registers an instance of its `YoutubeAudioSourceManager` unless
+explicitly excluded.
+
+First, create and register an instance of the supported `YoutubeAudioSourceManager` from the `youtube-source` package.
+```java
+AudioPlayerManager playerManager = new DefaultAudioPlayerManager();
+AudioSourceManager ytSourceManager = new dev.lavalink.youtube.YoutubeAudioSourceManager();
```
Comment (topi314) [inline README.md:189]:
```suggestion
YoutubeAudioSourceManager ytSourceManager = new dev.lavalink.youtube.YoutubeAudioSourceManager();
```

File: README.md [hunk 4]
```diff
@@ -177,7 +177,26 @@ Currently, the following clients are available for use:
     client enabled, age-restricted tracks are **not** guaranteed to play.
 
 ## Migration from Lavaplayer's built-in YouTube source
-This client is intended to be a drop-in replacement, however there are a couple of things to note:
+
+This client is intended as a direct replacement for Lavaplayer's `YoutubeAudioSourceManager`,
+which may be deprecated in more recent releases of Lavaplayer.
```
Comment (topi314) [inline README.md:182]:
```suggestion
which has been deprecated in a recent releases of Lavaplayer.
```

**Comment by devoxin** [comment]
As an aside, I think the use of `excludeSources` should be highlighted as a feature that only [lavalink-devs/lavaplayer](https://github.com/lavalink-devs/lavaplayer) supports currently. This isn't a standard feature found within all Lavaplayer forks so users may get confused if they try to use an overload that doesn't exist.

**Pull request merged**